src/rotation.zig

```diff
@@ -189,9 +196,18 @@ pub const RotationModes = struct {
 			return blocks.meshes.modelIndexStart(block) + @min(block.data, 5);
 		}
 
-		fn rotateZ(data: u16) u16 {
-			var neighbor: Neighbor = @enumFromInt(data);
-			return @intFromEnum(neighbor.rotateZ());
+		fn rotateZ(data: u16, angle: Degrees) u16 {
+			comptime var rotationTable: [24]u8 = undefined;
+			comptime for(0..6) |i| {
+				rotationTable[i] = i;
+			};
+			comptime for(1..4) |a| {
+				for(0..6) |i| {
+					const neighbor: Neighbor = @enumFromInt(rotationTable[i + 6*(a - 1)]);
+					rotationTable[i + 6*a] = neighbor.rotateZ().toInt();
+				}
+			};
+			return rotationTable[data + @as(u16, @intCast(@intFromEnum(angle)))*6];
```

Comment:
Should do bounds checking before plugging a random u16 into a table that only supports 6 values.

Otherwise worlds where the rotation mode of a block was different will crash.

In case of an incompatible value, you can just return 0 for the data.